app/server/utils/ussd/kenya_ussd_processor.py

```diff
@@ -97,8 +97,69 @@ def custom_display_text(menu: UssdMenu, ussd_session: UssdSession, user: User) -
             else:
                 return i18n_for(user, "{}.first".format(menu.display_key))
 
-        return None
-
+        if menu.name == 'directory_listing' or menu.name == 'send_token_reason':
+            items_per_menu = 8
 
+            usages = user.get_most_relevant_transfer_usage()
+            KenyaUssdProcessor.store_transfer_usage(ussd_session, usages)
+            ussd_session.set_data('usage_menu', 1)
+            ussd_session.set_data('usage_menu_max', math.floor(len(usages)/items_per_menu))
+            menu_options = KenyaUssdProcessor.create_usages_list(usages[:items_per_menu], user)
+            return i18n_for(
+                user, menu.display_key,
+                options=menu_options
+            )
 
+        if menu.name == 'directory_listing_other' or menu.name == 'send_token_reason_other':
```

Comment:
if you make the above change to pass in arbitrary data, you can test this better to see that all the different options work as you expect